src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -419,6 +448,22 @@ export class PatientViewPageStore {
         default: [],
     });
 
+    readonly samplesWithUniqueKeys = remoteData(
+        {
+            await: () => [this.samples, this.allSamplesForPatient],
+            invoke: () =>
+                Promise.resolve(
+                    this.samples.result.map(
```

Comment:
i'm not sure i get this operation here.  seems like you are just replacing

Reply:
Right, `this.samples` do not have unique keys, whereas `this.allSamplesForPatient` do. That's why we are replacing the samples here. We should probably make it more clear.